### PR TITLE
Tests solving

### PR DESCRIPTION
**Pull request by ekryukova-gd**

### CODE REVIEW

File: practice/3_python_testing/test_read_write_2_solved.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        sample.decode(encoding, errors='strict')
+        return True
+    except (UnicodeDecodeError, IOError):
+        return False
+
+
+def test_rw2():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        task_read_write_2_solved.rw2(temp_dir)
+        # with open(os.path.join(temp_dir, 'file1.txt'), 'rb') as result_file1:
```
Comment (mcoslet) [inline practice/3_python_testing/test_read_write_2_solved.py]: Why do you kept this commented code?

File: practice/3_python_testing/test_read_write_2_solved.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        #
+        #         suggestion1 = chardet.detect(f1)
+        #         suggestion2 = chardet.detect(f2)
+        #
+        #         assert suggestion1['encoding'].lower() == 'utf-8'
+        #         assert suggestion2['encoding'].lower() == 'cp1252'
+        path_file1 = os.path.join(temp_dir, 'file1.txt')
+        path_file2 = os.path.join(temp_dir, 'file2.txt')
+        with open(path_file1, 'r') as result_file1:
+            with open(path_file2, 'r') as result_file2:
```
Comment (mcoslet) [inline practice/3_python_testing/test_read_write_2_solved.py]: Why do you have nested `with` and why assert is in `with?

**Review by mcoslet — CHANGES_REQUESTED** [review]

File: practice/3_python_testing/test_read_write_2_solved.py (hunk 1 above)
Comment (ekryukova-gd) [reply practice/3_python_testing/test_read_write_2_solved.py]: corrected

File: practice/3_python_testing/test_read_write_2_solved.py (hunk 2 above)
Comment (ekryukova-gd) [reply practice/3_python_testing/test_read_write_2_solved.py]: corrected

**Review by mcoslet — APPROVED** [review]

**Pull request merged**